src/pyOpenMS/pxds/FeatureFinderAlgorithmMetaboIdent.pxd

```diff
@@ -47,6 +47,11 @@ cdef extern from "<OpenMS/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderAlgorithmMe
         FeatureFinderAlgorithmMetaboIdent_Row() nogil except +
         FeatureFinderAlgorithmMetaboIdent_Row(FeatureFinderAlgorithmMetaboIdent_Row) nogil except + #wrap-ignore
 
-
-  
-
+        String name
+        String formula
+        double mass
+        libcpp_vector[ Int ] charges
+        libcpp_vector[ doubles ] rts
+        libcpp_vector[ doubles ] rt_ranges
+        libcpp_vector[ doubles ] iso_distrib
+        
```

Comment:
probably possible with constructor:
r = FeatureFinderAlgorithmMetaboIdent_Row("caffeine", "C9H3", 200.0, [1], [34.4], null, null)
table = [ FeatureFinderAlgorithmMetaboIdent_Row("caffeine", "C9H3", 200.0, [1], [34.4], null, null), FeatureFinderAlgorithmMetaboIdent_Row("glucose", "C9H3", 200.0, [1], [34.4], null, null) ]

nicer ?:
r = ["caffeine", "C9H3", 200.0, [1], [34.4], null, null]
table = [ "caffeine", "C9H3", 200.0, [1], [34.4], null, null], ["glucose", "C9H3", 200.0, [1], [34.4], null, null] ]

how to get from tsv to table? pyx file needed?

Other question: can we include pandas?
would be probably best solution

Reply:
Other possibilities:
provide list of columns
 [ ["caffeine", "glucose"], ["C9H3", "C4H2"], [[200.0], [300.0]], [[1,2],[2]],  [[34.4], [45.4]], [null, null], [null, null]]

Reply:
https://tedboy.github.io/pandas/enhancingperf/enhancingperf1.html
´´´
In [10]: %%cython
   ....: cimport numpy as np
   ....: import numpy as np
   ....: cdef double f_typed(double x) except? -2:
   ....:     return x * (x - 1)
   ....: cpdef double integrate_f_typed(double a, double b, int N):
   ....:     cdef int i
   ....:     cdef double s, dx
   ....:     s = 0
   ....:     dx = (b - a) / N
   ....:     for i in range(N):
   ....:         s += f_typed(a + i * dx)
   ....:     return s * dx
   ....: cpdef np.ndarray[double] apply_integrate_f(np.ndarray col_a, np.ndarray col_b, np.ndarray col_N):
   ....:     assert (col_a.dtype == np.float and col_b.dtype == np.float and col_N.dtype == np.int)
   ....:     cdef Py_ssize_t i, n = len(col_N)
   ....:     assert (len(col_a) == len(col_b) == n)
   ....:     cdef np.ndarray[double] res = np.empty(n)
   ....:     for i in range(len(col_a)):
   ....:         res[i] = integrate_f_typed(col_a[i], col_b[i], col_N[i])
   ....:     return res
´´´

Reply:
Idee: pandas loads tsv -> converts to lists of lists -> runs pyx function that parses list of lists, builds rows and runs algorithm